ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/StatementExecutorTest.java

```diff
@@ -422,6 +448,100 @@ public void shouldEnforceReferentialIntegrity() {
 
   }
 
+  @Test
+  public void shouldTerminateClusterAndDeleteList() throws IOException {
+    // Given:
+    final List<String> deleteSources = Collections.singletonList("FOO");
+    final Map<String, Object> props = new HashMap<>();
+    props.put(TerminateCluster.DELETE_TOPIC_LIST_PARAM_NAME, deleteSources);
+    final Command command = new Command(
+        TerminateCluster.TERMINATE_CLUSTER_STATEMENT_TEXT,
+        props,
+        Collections.emptyMap());
+
+    // When:
+    testTerminateClusterForCommand(command, true);
+  }
+
+  @Test
+  public void shouldTerminateCluster() throws IOException {
```

Comment:
These tests, (even if you move them to CommandRunner) should be testing the functionality that is currently in `StatementExecutor`. i.e. it should be testing that it closes some stuff and invokes the `ClusterTerminator`.   That's all it needs to test. 

To do this, you'll need to inject the cluster terminator into the constructor. (Moving the KsqlConfig param from the constructor to the function call).

Then you can just pass in mocks for everything in this test class and make sure on terminate that the right things are getting closed and invoked.